### PR TITLE
Support BF16

### PR DESCRIPTION
**Pull request by findmyway**

### CODE REVIEW

**Comment by findmyway** [comment]
Hi @pabloferz ,the CIs are now all failed. Should we update the julia version to 1.10 consideribg it is the new LTS?

File: Project.toml [hunk 1]
```diff
@@ -17,6 +18,7 @@ PyCallExt = "PyCall"
 PythonCallExt = "PythonCall"
 
 [compat]
+BFloat16s = "0.5.0"
 Aqua = "0.8"
```
Comment (pabloferz) [inline Project.toml:22]:
```suggestion
Aqua = "0.8"
BFloat16s = "0.5.1"
```

File: src/DLPack.jl [hunk 2]
```diff
@@ -18,6 +18,7 @@ module DLPack
 ##  Dependencies  ##
 
 using Requires
+using BFloat16s: BFloat16
```
Comment (pabloferz) [inline src/DLPack.jl:21]:
```suggestion
using BFloat16s: BFloat16
using Requires
```

**Comment by pabloferz** [comment]
The main issue with CI is not the julia version but some changes upstream in how the DLPack API is handled in python. This requires a separate fix for supporting versioned `DLManagedTensor`s. I'll add a separate PR for that. In the meantime I think we can merge this. As long as you jax/pytorch/numpy version is not too recent, the package should still work.

**Review by pabloferz — DISMISSED** [review]

**Comment by pabloferz** [comment]
I'm unsure about the macOS PyCall failures, but I don't have a macOS machine to debug at the moment. Let me see what I can do about it.

**Comment by findmyway** [comment]
Thanks for the review! (Also congratulations!😄)

> I'm unsure about the macOS PyCall failures, but I don't have a macOS machine to debug at the moment. Let me see what I can do about it.

This might be related. https://github.com/JuliaGPU/CUDA.jl/issues/2579

Also I added some restrictions to the `jax` version to avoid an error similar to  https://github.com/jax-ml/jax/issues/15624

Overall, I think we can merge this first and wait for the release of julia@v1.10.8

**Comment by pabloferz** [comment]
>Also congratulations!😄
Thank you Jun!

v0.3.1 should be out now including these changes.

**Pull request merged**